core/src/main/java/com/graphhopper/routing/weighting/custom/CustomModelParser.java

```diff
@@ -409,45 +411,44 @@ private static SimpleCompiler createCompiler(long counter, Java.AbstractCompilat
         }
     }
 
-    static double findMaxSpeed(CustomModel customModel, double maxSpeed) {
-        return findMaxSpeed(customModel.getSpeed(), maxSpeed);
+    static double findMaxWithoutCheck(List<Statement> statements, double max) {
+        // we want to find the smallest speed that cannot be exceeded by any edge. the 'blocks' of speed statements
+        // are applied one after the other.
+        List<List<Statement>> blocks = splitIntoBlocks(statements);
+        for (List<Statement> block : blocks) max = findMaxForBlock(block, max);
+        return max;
     }
 
-    static double findMaxSpeed(List<Statement> speedStatements, final double maxSpeed) {
+    static double findMaxSpeed(List<Statement> statements, final double maxSpeed) {
         // throw an error if one of the limit statements won't possibly do anything
-        Optional<Statement> falseStatement = speedStatements.stream()
+        Optional<Statement> falseStatement = statements.stream()
                 .filter(st -> LIMIT.equals(st.getOperation()) && st.getValue() > maxSpeed)
                 .findFirst();
         if (falseStatement.isPresent())
             throw new IllegalArgumentException("Can never apply 'limit_to': " + falseStatement.get().getValue()
                     + " because maximum vehicle speed is " + maxSpeed);
 
```

Comment:
Why do you think would we need a maximum priority?

Reply:
No this checks simply checks that we do not do something like `speed: limit_to: 250` when max speed is only `140` anyway.

Reply:
The maximum speed is something physical that should not exceed practical values as ETA is affected, whereas the priority is something arbitrary / subjective - i.e. it can be 5 million without a problem. The limit of 1.5 is only in our EncodedValue (`encoder$priority`) which has not much to do with the general maximum priority here except that we will use 1.5 as default maximum if no other is found.

So basically when we start with 1.5 and unconditionally increase it with a factor of 10 and then do another unconditional `limit_to:5` should not be a problem (although illogical to have both in the same custom model)

Reply:
The following use case makes more sense and should work on the server-side:

```
{
  "priority" [{
      "if": "xy",
      "multiply_by": 5
  }, {
      "else": "",
      "multiply_by": 10
  }, {
      "if": "true",
      "limit_to": 5
  }
}
```

But it won't work if we use the "max storage" value of `encoder$priority` as the maximum possible value for the custom model priority.

Reply:
Ok, right. The check does not stop you from using `multiply_by: 5_000_000`, but you if we enable you can no longer first multiply with a large number and then limit to something above the storage limit.

But the thing I do not understand is why we should be allowed to do this for priority and not for speed. For example what if we want an electric bike (assuming storage max speed 30) like this:

```json
{
  "speed": [
      {
          "if": "surface != cobblestone",
          "multiply_by": 1.5
      },
      {
          "if": "true",
          "limit_to": 35
      }
  ]
}
```

Reply:
Thanks. This made me think :)
Currently the check would let you know that the limit_to will never have an effect because we use the storage max speed to reduce at the end of every speed calculation (probably I thought this is required because of LM). So we should just replace this storage-maxSpeed and use the result of findMaxSpeed if any. There are multiple advantages: we can increase the max speed via server-side config (nice!) and it would be better backward compatible (because every max speed reduction in storage could result in an exception for previously working custom model queries). 

Will change!

Reply:
Can you have a look into the last commit?